dataactcore/scripts/readZips.py

```diff
@@ -0,0 +1,136 @@
+                    zip_string = str(zip4_low).zfill(4)
+                    data_array[zip5 + zip_string] = {"zip5": zip5, "zip_last4": zip_string, "state_abbreviation": state,
+                                                     "county_number": county,
+                                                     "congressional_district_no": congressional_district}
+                # if the zip codes are different, we have to loop through and add each zip4 as a different object/key
+                else:
+                    i = zip4_low
+                    while i <= zip4_high:
+                        zip_string = str(i).zfill(4)
+                        data_array[zip5 + zip_string] = {"zip5": zip5, "zip_last4": zip_string,
+                                                         "state_abbreviation": state, "county_number": county,
+                                                         "congressional_district_no": congressional_district}
+                        i += 1
+            # catch entries where zip code isn't an int (12ND for example, ND stands for "no delivery")
+            except ValueError:
+                logger.error("error parsing entry: " + curr_row)
+
+            # cut the current line out of the chunk we're processing
+            curr_chunk = curr_chunk[line_size:]
+
+        # we want to do DB adding in large chunks so we can hopefully remove duplicates that are near each other
+        # in the file just by them having the same key in the dict
+        if len(data_array) > 100000:
+            logger.info("inserting next 100k+ records")
+            add_to_table(data_array, sess)
+            data_array.clear()
+
+    # add the final chunk of data to the DB
+    if len(data_array) > 0:
+        logger.info("adding last set of records for current file")
+        add_to_table(data_array, sess)
+        data_array.clear()
+
+
+def main():
+    sess = GlobalDB.db().session
+
+    # delete old values in case something changed and one is now invalid
+    sess.query(Zips).delete(synchronize_session=False)
+    sess.commit()
+
+    if CONFIG_BROKER["use_aws"]:
+        s3connection = boto.s3.connect_to_region(CONFIG_BROKER['aws_region'])
+        s3bucket = s3connection.lookup(CONFIG_BROKER['sf_133_bucket'])
+        zip_folder = CONFIG_BROKER["zip_folder"] + "/"
+        for key in s3bucket.list(prefix=zip_folder):
+            if key.name != zip_folder:
+                zip_4_file_path = key.generate_url(expires_in=600)
+                parse_zip4_file(urllib.request.urlopen(zip_4_file_path), sess)
+    else:
+        base_path = os.path.join(CONFIG_BROKER["path"], "dataactvalidator", "config", CONFIG_BROKER["zip_folder"])
+        file_list = [f for f in os.listdir(base_path)]
+        for file in file_list:
+            parse_zip4_file(open(os.path.join(base_path, file)), sess)
+
+    logger.info("Zipcode script complete")
+
+
+if __name__ == '__main__':
+    with create_app().app_context():
```

Comment:
Since this script will be run on it's own, do we need to call `configure_logging`?